src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,75 @@
+import { useAppDispatch, useAppSelector } from '../../app/store';
 
 export const TodoModal: React.FC = () => {
+  const [currentUser, setCurrentUser] = useState<User | null>(null);
+  const [isLoaderActive, setIsLoaderActive] = useState(false);
+
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  useEffect(() => {
+    setIsLoaderActive(true);
+
+    if (currentTodo) {
+      getUser(currentTodo.userId)
+        .then(setCurrentUser)
+        .finally(() => setIsLoaderActive(false));
+    }
+  }, [currentTodo?.userId]);
+
+  const closeModalHandler = () => {
+    dispatch(currentTodoSlice.actions.delete());
+  };
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {isLoaderActive ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              {currentTodo?.id ? `Todo #${currentTodo.id}` : ''}
+            </div>
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={closeModalHandler}
+            />
+          </header>
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+          <div className="modal-card-body">
+            <p className="block" data-cy="modal-title">
+              {currentTodo?.title}
+            </p>
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+            <p className="block" data-cy="modal-user">
+              {currentTodo?.completed ? (
+                <strong className="has-text-success">Done</strong>
+              ) : (
+                <strong className="has-text-danger">Planned</strong>
+              )}
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+              {' by '}
 
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+              <a href={currentUser?.email}>{currentUser?.name}</a>
```

Comment:
Ensure that `currentUser` is not null before accessing its properties like `email` and `name`. Consider adding a conditional check or a fallback to handle cases where `currentUser` might be null.